src/java/org/jivesoftware/openfire/pep/IQPEPHandler.java

```diff
@@ -278,7 +279,14 @@ public IQHandlerInfo getInfo() {
      * in the server's disco#info result (as per section 4 of XEP-0163).
      */
     public Iterator<String> getFeatures() {
-        return XMPPServer.getInstance().getPubSubModule().getFeatures(null, null, null);
+        Iterator<String> it = XMPPServer.getInstance().getPubSubModule().getFeatures(null, null, null);
+        ArrayList<String> features = new ArrayList<String>();
+        for (;it.hasNext();) {
```

Comment:
Just a very minor remark: Personally I like a while loop here better for readability. But eventually it doesn't really matter.